docs/software-development-kits/java/middleware.mdx

```diff
@@ -0,0 +1,133 @@
+```java
+@ComponentScan("com.aserto")
+```
+
+## Configuration
+
+The authorization filter accepts the fallowing configuration parameters:
+
+| Parameter name | Default Value | Description |
+| -------------- | ------------- | ----------- |
+| aserto.authorization.enabled | true | Enable/disable the authorization filter |
+| aserto.authorization.host | localhost | The hostname of the authorizer service |
+| aserto.authorization.port | 8282 | The port of the authorizer service |
+| aserto.authorizer.insecure | false | Use insecure connection to the authorizer service |
+| aserto.authorizer.cert.path | N/A | The path to the authorizer service certificate, usefull when using local topaz with self signed certificates |
+| aserto.authorizer.apiKey | N/A | The API key to use when connecting to the authorizer service |
+| aserto.tenantId | N/A | The tenant ID to use when connecting to the authorizer service |
+| aserto.authorizer.policyName | N/A | The policy name used by the authorizer |
+| aserto.authorizer.policyLabel | N/A | The policy label used by the authorizer |
+| aserto.authorizer.decision | N/A | The decision used by the authorizer |
+| logging.level.com.aserto | N/A | The log level for the Aserto Middleware |
+
+
+## Identity
+To determine the identity of the user, the middleware can be configured to use a JWT token or a claim using the IdentityMapper.
+All you have to do is create a bean that return a InstanceMapper interface and the middleware will use it to determine the identity of the user.
+
+E.g.
+- use a JWT token from the `Authorization` header:
+```java
+    @Bean
+    public IdentityMapper identityDiscoverer() {
+        Extractor headerExtractor = new HeaderExtractor("Authorization");
+        return new JwtIdentityMapper(headerExtractor);
+    }
+```
+
+- use the "sub" claim from the JWT token in the `Authorization` header:
+```java
+    @Bean
+    public IdentityMapper identityDiscoverer() {
+        Extractor headerExtractor = new AuthzHeaderExtractor("Authorization", "sub");
+        return new SubjectIdentityMapper(headerExtractor);
+    }
+```
+
+You can also implement your own IdentityMapper by implementing the `IdentityMapper` interface.
+
+## Policy path
+By default, when computing the policy path, the middleware:
+* converts all slashes to dots
+* converts any path parameters to `__<param_name>`
+* converts uppercase characters in the URL path to lowercase
+
+As with the identity contest, you can implement your own PolicyPathMapper by implementing the `PolicyPathMapper`
+interface and creating a bean that returns it.
+
+E.g.
+```java
+    @Bean
+    public PolicyMapper policyMapperDiscoverer() {
+        return new MyCustomPolicyPathMapper(handlerMapping);
+    }
+```
+
+## Resource
+A resource can be any structured data that the authorization policy uses to evaluate decisions.
+By default, the middleware does not include a resource in authorization calls.
```

Comment:
I think the default behavior should be that all path parameters are sent in the resource context.
For example, if the route being called is `/api/users/{user_id}` then, by default, the middleware should extract the `user_id` parameter and send it in the resource context.
In other words a `GET /api/users/1234` request should result in the following resource context:
```json
{
  "user_id": 1234
}
```

Reply:
that is consistent with how the rest of our middlewares work

Reply:
Added the `PathParamsResourceMapper` to the [aserto-spring](https://github.com/aserto-dev/aserto-spring/commit/73739d8660f2ab5d93dc0d3f237e43026cf52276) and make it the default mapper.
I updated the docs to reflect this.